critiquebrainz/frontend/templates/macros.html

```diff
@@ -220,3 +220,41 @@ <h4 style="margin-bottom:0;">{{ _('Rate this {}:'.format(entity_type_readable))
         -
     {% endif %}
 {% endmacro %}
+
+{% macro display_pagination(current_page, count, limit, page_info, parameter) %}
+  {% set total_pages = (count/limit)|round(method='ceil')|int %}
+  <div class="col-md-12 text-center">
+    <nav aria-label="Page navigation">
+      <ul class="pagination">
+
+        <li class="previous {{'disabled' if current_page <= 1 }}"><a aria-label="Previous" href="{{ url_for(page_info, page=page-1, **parameter) }}">&larr; {{ _('Previous') }}</a></li>
+        
+        <li {{ "class=active" if current_page == 1 }}><a href="{{ url_for(page_info, page=1,  **parameter) }}">{{ 1 }}</a></li>
+
+        {% if current_page > 5 %}
+          <li class="disabled"><a>{{ '...' }}</a></li>
+        {% endif %}
+
+        {% if current_page == 5 %}
+          <li><a href="{{ url_for(page_info, page=2, **parameter) }}">{{ 2 }}</a></li>
+        {% endif %}
+
+        {% for page_index in range([2,current_page-2]|max, [total_pages,current_page+3]|min) %}
+          <li {{ "class=active" if current_page == page_index }}><a href="{{ url_for(page_info, page=page_index, **parameter) }}">{{ page_index }}</a></li>
+        {% endfor %}
+
+        {% if current_page == (total_pages-4) %}
+          <li><a href="{{ url_for(page_info, page=(total_pages-1), **parameter) }}">{{ total_pages-1 }}</a></li>
+        {% endif %}
+
+        {% if (current_page < (total_pages - 4)) %}
+          <li class='disabled'><a>{{ '...' }}</a></li>
+        {% endif %}
+
+        <li {{ "class=active" if current_page == total_pages }}><a href="{{ url_for(page_info, page=total_pages, **parameter) }}">{{ total_pages }}</a></li>
+
+        <li class="next {{'disabled' if (current_page-1) >= total_pages }}"><a aria-label="Next" href="{{ url_for(page_info, page=page+1, **parameter) }}">{{ _('Next') }} &rarr;</a></li>
```

Comment:
While testing, I saw that if the user navigates to the last page. The next button is still enabled. This condition should probably be using `current_page` instead of `current_page - 1`.